api_app/analyzers_manager/observable_analyzers/greynoiseintel.py

```diff
@@ -0,0 +1,58 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+from greynoise import GreyNoise
+
+from api_app.analyzers_manager import classes
+from api_app.exceptions import AnalyzerRunException
+from tests.mock_utils import MockResponse, if_mock_connections, patch
+
+
+class GreyNoiseAnalyzer(classes.ObservableAnalyzer):
+    def set_params(self, params):
+        self.api_version = params.get("greynoise_api_version", "v3")
+        self.max_records_to_retrieve = int(params.get("max_records_to_retrieve", 500))
+
+    def run(self):
```

Comment:
an overall question. How the 4xx and 5xx status codes are handled by this library? What happens if one of them is triggered?

Reply:
They are going to be returned as exceptions, so I've added in handling for that, so it returns the error back to the user.  Let me know if you'd like to see something different here.